### PR TITLE
[WIP] Improving benchmarks

### LINKED ISSUE #782: A better way to benchmark the speed tests
The benchmark tests which are in use now can be improved much more. Importing arviz into the benchmark tests throws the following error: 

Screenshot from 2019-08-01 21-45-25

The workaround right now is to include only the jitted methods in `benchmarks.py`. It is indicative of the extent of the speedup but isn't very accurate.

### PR DESCRIPTION
**Pull request by nitishp25**
## Description

Fixes #782 and builds on top of the work done in #1088.

I've moved `_fast_kde` and `_fast_kde_2d` to `kde_utils` and `get_bins` to `stats_utils` to resolve cyclic import issues occuring due to import of `histogram` for the `_fast_kde`.

I've also modified the benchmarks and will cover more functions soon. Currently, the `stats_variance_2d` benchmark does not work with Numba disabled but I couldn't figure out why (the function still works with Numba disabled).

cc @OriolAbril, @ahartikainen

## Checklist

- [x] Follows [official](https://github.com/arviz-devs/arviz/blob/master/CONTRIBUTING.md#pull-request-checklist) PR format
- [ ] Includes new or updated tests to cover the new feature
- [x] Code style correct (follows pylint and black guidelines)
- [ ] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/master/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

**Comment by OriolAbril** [comment]
Can you rebase the changes to current master? it looks like the first two should not be included in this PR and clutter the files to be reviewed :/

**Comment by nitishp25** [comment]
> Can you rebase the changes to current master? it looks like the first two should not be included in this PR and clutter the files to be reviewed :/

Shouldn't the `arviz-devs/benchmarks` branch be also rebased with the master?

**Comment by OriolAbril** [comment]
Ohhhh, I completely overlooked this, I though this was built on top of #1088 to merge into master, not to merge into benchmarks PR. Yes, `benchmarks` has to be rebased on master and then this branch has to be rebased on benchmarks for this to work.

File: arviz/plots/plot_utils.py [hunk 1]
```diff
@@ -99,48 +98,6 @@ def _scale_fig_size(figsize, textsize, rows=1, cols=1):
...
-    # The Freedman-Diaconis histogram bin estimator.
-    iqr = np.subtract(*np.percentile(values, [75, 25]))  # pylint: disable=assignment-from-no-return
-    bins_fd = 2 * iqr * values.size ** (-1 / 3)
-
-    width = round(np.max([1, bins_sturges, bins_fd])).astype(int)
-
-    return np.arange(x_min, x_max + width + 1, width)
-
-
 def _sturges_formula(dataset, mult=1):
```
Comment (OriolAbril) [inline arviz/plots/plot_utils.py:101]: I am not sure about what functions use this, but it looks like it should be moved to `stats_utils` like `get_bins`

**Comment by nitishp25** [comment]
> Ohhhh, I completely overlooked this, I though this was built on top of #1088 to merge into master, not to merge into benchmarks PR. Yes, `benchmarks` has to be rebased on master and then this branch has to be rebased on benchmarks for this to work.

I was whether to make this PR to `master` or `benchmarks`. Is this more inconvinient than making a PR to master?

File: arviz/stats/stats_utils.py [hunk 2]
```diff
@@ -7,6 +7,8 @@
 import numpy as np
 import pandas as pd
 from scipy.fftpack import next_fast_len
+from scipy.signal import convolve
+from scipy.signal.windows import gaussian
```
Comment (OriolAbril) [inline arviz/stats/stats_utils.py:11]: are these needed? maybe they are used by fast_kde?

**Comment by OriolAbril** [comment]
> I was confused whether to make this PR to `master` or `benchmarks`. Is this more inconvinient than making a PR to master?

Whatever is less work for you will be best.

File: arviz/plots/plot_utils.py (hunk 1 above)
Comment (nitishp25) [reply arviz/plots/plot_utils.py:101]: Currently, only `rank_plot` uses this

File: arviz/stats/stats_utils.py (hunk 2 above)
Comment (nitishp25) [reply arviz/stats/stats_utils.py:11]: Yes, there are a few more unsed imports. I'll remove them

**Comment by nitishp25** [comment]
> Whatever is less work for you will be best.

I think this way you would have to always rebase this branch with master everytime you want to review so it's better if I close this and make this PR to master? :)

**Pull request closed without merging**